packages/react-compose/src/next/README.md

```diff
@@ -0,0 +1,315 @@
+
+### But why is extend important?
+
+Fluent UI components should have 2 separate layers; base components which are unstyled, and styled versions of those components. This gives developers 2 layers to use; use the styled component when building M365 experiences. Extend the unstyled component (e.g. `ButtonBase`) to use a different styling approach, or build a new, slightly augmented component extending the styled version (e.g. `Button`).
+
+## Getting started writing components
+
+### Write and review a spec
+
+Know what you're building. Do the research on the following:
+
+- What is the right component name? Consult guidance in Open UI.
+- What props are accepted? Are there standards in Open UI
+- What slots should be configurable?
+- How does customized styling work?
+
+### Build a base component
+
+A base component defines the render dom heirarchy, state management, and accessible behaviors. It does not provide styling. It does not attach dependencies. These are done in a separate layer, so that customers can always fall back to extending the base, should they wish to provide their own styling or subcomponent dependencies.
+
+Use `compose` to create the extendable base component. It takes in a render function and options:
+
+```jsx
+const MyComponentBase = compose<TProps, TState>(
+  (state, options) => {
+    return <div />;
+  }, {
+    displayName: 'MyComponentBase',
+    // ...
+  });
+```
+
+## Supporting component slots
+
+The root element should always be configurable by the customer using the `as` prop.
+
+Additionally components often have more than a root element; these additional subcomponents should be configurable by the customer as well. We customize these using slot props.
+
+Use the `getSlots` helper to parse out `as` prop, slots and native props. By default, it will return a `root` slot for you.
+
+```
+const MyComponentBase = compose((state, options) => {
+  const { slots, slotProps } = getSlots(state, options);
+
+  return <slots.root {...slotProps.root} />;
+}, {
+  displayName: 'MyComponentBase',
+  defaultProps: {
+    as: 'div',
+    icon: { as: FooComponent }
+  }
+]});
+```
+
+Add an `icon` slot:
+
+```jsx
+const MyComponentBase = compose(
+  (state, options) => {
+    const { slots, slotProps } = getSlots(state, options);
+
+    return (
+      <slots.root {...slotProps.root}>
+        <slots.icon {...slotProps.icon} />
+      </slots.root>
+    );
+  },
+  {
+    // Instruct merging on which props are shorthand.
+    shorthandProps: ['icon'],
+
+    defaultProps: {
+      // Define the default props for the icon.
+      icon: { as: 'span' },
+    },
+  },
+);
+```
+
+Note: the `shorthandProps` array option is required for compose to apply shorthand normalization logic for proper default props merging. (Because a shorthand prop can be a string, object, or JSX, you need to apply custom merge logic for these.)
```

Comment:
we should have a section in README for shorthand/slot prop with more details on how to use